wandb/sdk/launch/builder/build.py

```diff
@@ -334,20 +336,59 @@ def get_requirements_section(launch_project: LaunchProject, builder_type: str) -
+        base_path = pathlib.Path(launch_project.project_dir)
+        # If there is a requirements.txt at root of build context, use that.
+        if (base_path / "requirements.txt").exists():
             requirements_files += ["src/requirements.txt"]
             pip_install_line = "pip install -r requirements.txt"
-        elif launch_project.project_dir is not None and os.path.exists(
-            os.path.join(launch_project.project_dir, "requirements.frozen.txt")
-        ):
-            # if we have frozen requirements stored, copy those over and have them take precedence
-            requirements_files += ["src/requirements.frozen.txt", "_wandb_bootstrap.py"]
+        # Elif there is pyproject.toml at build context, convert the dependencies
+        # section to a requirements.txt and use that.
+        elif (base_path / "pyproject.toml").exists():
+            tomli = get_module("tomli")
+            if tomli is None:
+                wandb.termwarn(
+                    "pyproject.toml found but tomli could not be loaded. To "
+                    "install dependencies from pyproject.toml please run "
+                    "`pip install tomli` and try again."
+                )
+            else:
+                # First try to read deps from standard pyproject format.
+                with open(base_path / "pyproject.toml", "rb") as f:
+                    contents = tomli.load(f)
+                project_deps = [
+                    str(d) for d in contents.get("project", {}).get("dependencies", [])
+                ]
+                if project_deps:
+                    with open(base_path / "requirements.txt", "w") as f:
+                        f.write("\n".join(project_deps))
+                    requirements_files += ["src/requirements.txt"]
+                    pip_install_line = "pip install -r requirements.txt"
+                # If not found, check to see if we can install with poetry.
+                else:
+                    if "poetry" in contents.get("tool"):
+                        requirements_files += ["src/pyproject.toml"]
+                        pip_install_line = (
+                            "pip install poetry && "
+                            "poetry config virtualenvs.create false --local && "
+                            "poetry install --no-root"
+                        )
+        # Else use frozen requirements from wandb run.
+        if not pip_install_line and (base_path / "requirements.frozen.txt").exists():
+            requirements_files += [
+                "src/requirements.frozen.txt",
+                "_wandb_bootstrap.py",
+            ]
             pip_install_line = (
                 _parse_existing_requirements(launch_project)
                 + "python _wandb_bootstrap.py"
             )
+
+        if not pip_install_line:
+            wandb.termwarn("No python dependency files found, only installing wandb.")
+            pip_install_line = "pip install --upgrade wandb"
```

Comment:
@KyleGoyette w.r.t. requiring dependency files, look at this change and lmk what you think